src/OrbitDB.js

```diff
@@ -48,8 +47,8 @@ class OrbitDB {
     this.storage = options.storage
     this._directConnections = {}
 
-    this.caches = { }
-    this.caches[this.directory] = options.cache
+    this.caches = {}
+    this.caches[this.directory] = { cache: options.cache, handlers: new Set() }
```

Comment:
This poses a semantic issue no? Does it make sense that all empty-handler caches should be closed if this one is by default empty?